apps/src/code-studio/components/exportDialogRedux.js

```diff
@@ -0,0 +1,38 @@
+// Action types
+
+const SHOW_EXPORT_DIALOG = 'exportDialog/SHOW_EXPORT_DIALOG';
+const HIDE_EXPORT_DIALOG = 'exportDialog/HIDE_EXPORT_DIALOG';
+
+// Reducer
+
+const initialState = {
+  isOpen: false
+};
+
+export default function reducer(state = initialState, action) {
+  switch (action.type) {
+    case SHOW_EXPORT_DIALOG:
+      return {
+        ...state,
+        ...initialState,
+        isOpen: true
```

Comment:
This seems like overkill.

Reply:
overkill meaning using redux to track whether the dialog is open? I agree that it is odd, but I started by following the share dialog and reduced most of the code away. I decided to keep this part because it allows us to use the dialog in a more "modeless" fashion (where you could close it while the export is underway, and open it back up later). I'm still in discussions with @ryansloan over the UI metaphor we'd use to enable that.

Reply:
I specifically meant `...state, ...initialState, isOpen: true` when there's only the one property here.  I think I see the spirit of the thing though. :grin: